apps/src/netsim/PubSubService.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @overview Wrapped pub/sub service client APIs (like Pusher)
+ */
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+
+ maxlen: 90,
+ maxparams: 3,
+ maxstatements: 200
+ */
+/* global Pusher */
+'use strict';
+
+var NetSimLogger = require('./NetSimLogger');
+var PubSubChannel = require('./PubSubChannel');
+var netsimGlobals = require('./netsimGlobals');
+
+/**
+ * JavaScript interface for a publish/subscribe service provider.
+ * @param {string} applicationKey
+ * @interface
+ */
+var PubSubService = exports;
+
+/**
+ * Create an API instance appropriate to the current configuration.
+ * @returns {PubSubService}
+ */
+PubSubService.create = function () {
+  var pubSubConfig = netsimGlobals.getPubSubServiceInfo();
```

Comment:
To minimize knowledge of globals and improve configurability, I think it might be better to take the pubSubConfig as a parameter.